### PR TITLE
[VMD] Fix observeAsState for instance properties

### PR DESCRIPTION
**Pull request by npresseault**
## Description
The exception thrown was:
`Fatal Exception: java.lang.IllegalArgumentException: Callable expects 0 arguments, but 1 were provided.`

Since #158 it was no longer possible to call `observeAsState` using a `KProperty0` (instance property). We now properly support `KProperty0` (instance property) and `KProperty1` (class property with a receiver argument)

For reference
```
val kPropReceiver: KProperty0<Boolean> = viewModel::isHidden 
val kProp: KProperty1<VMDViewModel, Boolean> = VMDViewModel::isHidden
```

`KProperty2`: extension function with a different receiver will not be supported.

## Motivation and Context
We want to be able to use both property types

## How Has This Been Tested?
Tested with both property types

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: trikot-viewmodels-declarative-flow/compose-flow/src/main/kotlin/com/mirego/trikot/viewmodels/declarative/compose/extensions/ViewModelExtensions.kt [hunk 1]
```diff
@@ -133,3 +135,12 @@ inline fun <T, S> S.asState(
     }
     return state
 }
+
+@Suppress("UNCHECKED_CAST")
+private  fun <T, V> KProperty<V>.callGetter(receiver: T): V =
```
Comment (etienneasselin) [inline trikot-viewmodels-declarative-flow/compose-flow/src/main/kotlin/com/mirego/trikot/viewmodels/declarative/compose/extensions/ViewModelExtensions.kt:140]: two spaces between private and fun

**Review by etienneasselin — APPROVED** [review]

File: trikot-viewmodels-declarative-flow/compose-flow/src/main/kotlin/com/mirego/trikot/viewmodels/declarative/compose/extensions/ViewModelExtensions.kt (hunk 1 above)
Comment (npresseault) [reply trikot-viewmodels-declarative-flow/compose-flow/src/main/kotlin/com/mirego/trikot/viewmodels/declarative/compose/extensions/ViewModelExtensions.kt:140]: ktlint seems not to run on those modules, I'll fix this.

Comment (npresseault) [reply trikot-viewmodels-declarative-flow/compose-flow/src/main/kotlin/com/mirego/trikot/viewmodels/declarative/compose/extensions/ViewModelExtensions.kt]: Ok so I setupped ktlint properly, lots of unrelated changes, sorry for that.

File: trikot-viewmodels-declarative/compose/src/main/kotlin/com/mirego/trikot/viewmodels/declarative/compose/extensions/ViewModelExtensions.kt [hunk 2]
```diff
@@ -111,3 +113,11 @@ inline fun <T, S> S.asState(
     }
     return state
 }
+
+@Suppress("UNCHECKED_CAST")
+private fun <T, V> KProperty<V>.callGetter(receiver: T): V =
```
Comment (ccipelli-mirego) [inline trikot-viewmodels-declarative/compose/src/main/kotlin/com/mirego/trikot/viewmodels/declarative/compose/extensions/ViewModelExtensions.kt:118]: 👍

**Review by ccipelli-mirego — COMMENTED** [review]
🤙

**Pull request merged**